src/libcore/num/mod.rs

```diff
@@ -2037,6 +2037,119 @@ assert_eq!((-a).rem_euclid(-b), 1);
             }
         }
 
+        doc_comment! {
+            concat!("Returns the logarithm of the number with respect to an arbitrary base.
+
+Returns `None` if the number is lower than 1.
+
+This method may not be optimized owing to implementation details;
+`self.log2()` can produce results more efficiently for base 2, and
+`self.log10()` can produce results more efficiently for base 10.
+
+# Examples
+
+```
+#![feature(int_log)]
+
+let five = 5", stringify!($SelfT), ";
+
+// log5(5) == 1
+let result = five.log(5);
+
+assert_eq!(result, Some(1));
+```"),
+            #[unstable(feature = "int_log", issue = "70887")]
+            #[must_use = "this returns the result of the operation, \
+                          without modifying the original"]
+            #[inline]
+            pub fn log(self, base: Self) -> Option<Self> {
+                let logb2 = |x: Self| {
+                    let log = (1 as Self).leading_zeros() - x.leading_zeros();
+                    log as Self
+                };
+
+                if self <= 0 || base <= 1 {
+                    None
+                } else if base > self {
+                    Some(0)
```

Comment:
This check looks unnecessary; the code below already returns `Some(0)` if `base` > `self`.